File: packages/php/src/Resources/Analytics.php
<?php

declare(strict_types=1);

namespace MissionInbox\Resources;

use MissionInbox\Http\Client;

/**
 * The `analytics` resource. Access via `$mi->analytics`.
 */
final class Analytics
{
    public function __construct(private readonly Client $http) {}

    /** Account-wide overview: this month + last month send counts, domain/mailbox counts. */
    public function getOverview(): array
    {
        return $this->http->request('GET', '/api/analytics/overview');
    }

    /**
     * Time-series data for send/receive activity.
     *
     * @param array{
     *   granularity: 'hourly'|'daily'|'monthly'|'yearly',
     *   startDate: string,
     *   endDate: string,
     *   counters?: array<'incoming'|'outgoing'|'bounces'|'spam'|'held'>
     * } $params
     */
    public function getActivityGraph(array $params): array
    {
        $query = [
            'granularity' => $params['granularity'],
            'startDate' => $params['startDate'],
            'endDate' => $params['endDate'],
        ];
        if (!empty($params['counters'])) {
            $query['counters'] = $params['counters'];
        }
        return $this->http->request('GET', '/api/analytics/activity-graph', query: $query);
    }
}

File: packages/python/src/missioninbox/resources/analytics.py
"""The `analytics` resource. Access via `mi.analytics`."""

from __future__ import annotations

from typing import Any, Optional

from .._http import HttpClient


class Analytics:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def get_overview(self) -> dict[str, Any]:
        """Account-wide overview: send counts, domain / mailbox counts."""
        return self._http.request("GET", "/api/analytics/overview")

    def get_activity_graph(
        self,
        *,
        granularity: str,
        start_date: str,
        end_date: str,
        counters: Optional[list[str]] = None,
    ) -> dict[str, Any]:
        """Time-series data for send/receive activity.

        Args:
            granularity: One of ``hourly``, ``daily``, ``monthly``, ``yearly``.
            start_date: ISO 8601 timestamp.
            end_date: ISO 8601 timestamp.
            counters: Optional filter — any of ``incoming``, ``outgoing``,
                ``bounces``, ``spam``, ``held``.
        """
        query: dict[str, Any] = {
            "granularity": granularity,
            "startDate": start_date,
            "endDate": end_date,
        }
        if counters:
            query["counters"] = counters
        return self._http.request("GET", "/api/analytics/activity-graph", query=query)
